swarm/storage/mru/resource.go

```diff
@@ -245,45 +238,55 @@ func (h *Handler) SetStore(store *storage.NetStore) {
 // It implements the storage.ChunkValidator interface
 func (h *Handler) Validate(addr storage.Address, data []byte) bool {
 
+	if data[0] == 0 && data[1] == 0 && len(data) > common.AddressLength {
+		//metadata chunk
+		rootAddr, _ := metadataHash(data)
+		valid := bytes.Equal(addr, rootAddr)
+		if !valid {
+			log.Warn("Invalid root metadata chunk")
+		}
+		return valid
+	}
+
```

Comment:
Validator now also checks metadata chunks.
To be fixed: should check `len(data) > 2` before looking at the first two bytes.

Reply:
The metadata chunk is still content addressed, however looking at the first two bytes at least saves calculating a hash if the incoming chunk doesn't look like a metadata chunk at first sight.

I am fine either way. We can do a hash and if it matches we mark as valid. What option would you choose?

Reply:
It is still content addressed, but the two bytes were a quick way to determine if an incoming chunk looked like a metadata chunk and thus it was worthy of us to hash it and see if it was good.

I am fine removing this, but let's take it to #704, where encoding is easier to manipulate.

Reply:
ah yes the update chunks are int16 first two of course.